pkg/tsdb/influxdb/influxdb.go

```diff
@@ -73,18 +74,19 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 		}
 
 		model := &models.DatasourceInfo{
-			HTTPClient:    client,
-			URL:           settings.URL,
-			DbName:        database,
-			Version:       version,
-			HTTPMode:      httpMode,
-			TimeInterval:  jsonData.TimeInterval,
-			DefaultBucket: jsonData.DefaultBucket,
-			Organization:  jsonData.Organization,
-			MaxSeries:     maxSeries,
-			SecureGrpc:    true,
-			Token:         settings.DecryptedSecureJSONData["token"],
-			Timeout:       opts.Timeouts.Timeout,
+			HTTPClient:      client,
+			URL:             settings.URL,
+			DbName:          database,
+			Version:         version,
+			HTTPMode:        httpMode,
+			TimeInterval:    jsonData.TimeInterval,
+			DefaultBucket:   jsonData.DefaultBucket,
+			Organization:    jsonData.Organization,
+			MaxSeries:       maxSeries,
+			SecureGrpc:      true,
+			Token:           settings.DecryptedSecureJSONData["token"],
+			Timeout:         opts.Timeouts.Timeout,
+			ConcurrentQuery: cfg.ConcurrentQueryCount,
```

Comment:
This goes against the goal of decoupling core plugins since external plugins don't have direct access to the Grafana config. To add this, add it to the config map that is forwarded for all plugins:

https://github.com/grafana/grafana/blob/main/pkg/plugins/envvars/envvars.go#L102

Then you can obtain it from the request in the Query method like:

```
func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
	concurrentQueryCount := req.PluginContext.GrafanaConfig.Get("key")
```

As a helper, you can expose the key you are using in the SDK and even a function to easily retrieve it (this is optional), like it's done for the AppURL:

https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/config.go#L133

Reply:
Hey thanks a lot for the review. I was expecting such a review to be honest :) 
I will try to implement what you suggest but things are getting funky here.
Looks like I need to expose but setting [in here](https://github.com/grafana/grafana/blob/1caaa56de07a46103db2cfd787737a5491bf1db5/pkg/services/pluginsintegration/config/config.go#L30-L53) (ProvideConfig in pkg/services/pluginsintegration/config/config.go) to be able to pass it the place you point `GetConfigMap`.
Is this right or am I making this more complicated than it should be?
I am asking this because this is a simple setting that plugin should receive. If we’ll start to do such thing for every setting that plugin needs there will be a mess.

Reply:
Exposing the key: https://github.com/grafana/grafana-plugin-sdk-go/pull/875

Reply:
This is now handled as you suggested. Thanks for the explanation and details. It helped a lot.